res/css/views/rooms/_TopUnreadMessagesBar.pcss

```diff
@@ -22,60 +22,73 @@ limitations under the License.
     top: 24px;
     right: 24px;
     width: 38px;
-}
 
-.mx_TopUnreadMessagesBar::after {
-    content: "";
-    position: absolute;
-    top: -8px;
-    left: 10.5px;
-    width: 4px;
-    height: 4px;
-    border-radius: 16px;
-    background-color: $secondary-accent-color;
-    border: 6px solid $accent;
-    pointer-events: none;
-}
+    &::after {
+        content: "";
+        position: absolute;
+        top: -8px;
+        left: 10.5px;
+        width: 4px;
+        height: 4px;
+        border-radius: 16px;
+        background-color: $secondary-accent-color;
+        border: 6px solid $accent;
+        pointer-events: none;
+    }
 
-.mx_TopUnreadMessagesBar_scrollUp {
-    height: 38px;
-    border-radius: 19px;
-    box-sizing: border-box;
-    background: $background;
-    border: 1.3px solid $muted-fg-color;
-    cursor: pointer;
-}
+    /* Define common declarations. Their values are specified below. */
+    .mx_TopUnreadMessagesBar_scrollUp,
+    .mx_TopUnreadMessagesBar_markAsRead {
```

Comment:
what is the reason for nesting these where they were not before?

Reply:
To avoid repeating the same declarations, and to make the structure compact. Do you rather want to write like this?

````
    .mx_TopUnreadMessagesBar_scrollUp {
        background: var(--background);
        width: var(--width);
        height: var(--height);
        border: var(--border);
        border-radius: var(--borderRadius);

        --background: $background;
        --border: 1.3px solid $muted-fg-color;

        &::before {
            content: "";
            position: absolute;
            background: var(--background);
            width: var(--width);
            height: var(--height);
            mask-repeat: no-repeat;
            mask-position: center;
            mask-image: var(--maskImage);
            mask-size: var(--maskSize);

            --background: $muted-fg-color;
        }
    }

    .mx_TopUnreadMessagesBar_markAsRead {
        background: var(--background);
        width: var(--width);
        height: var(--height);
        border: var(--border);
        border-radius: var(--borderRadius);

        --background: $background;
        --border: 1.3px solid $muted-fg-color;

        &::before {
            content: "";
            position: absolute;
            background: var(--background);
            width: var(--width);
            height: var(--height);
            mask-repeat: no-repeat;
            mask-position: center;
            mask-image: var(--maskImage);
            mask-size: var(--maskSize);

            --background: $muted-fg-color;
        }
    }
````

It works, but it's not really efficient.

Reply:
Also, there seems to me that there has not been a specific reason either that those selectors were not combined, checking a commit like https://github.com/matrix-org/matrix-react-sdk/pull/2345/commits/ac9902e52a704f52e508757fef7cf8c9b517f11b#diff-21b2361c0ea031da22f3680c5912c17ccc3d86c5bad901dd533f004cde3efa32 of https://github.com/matrix-org/matrix-react-sdk/pull/2345 and https://github.com/matrix-org/matrix-react-sdk/pull/4159/commits/07348a65ad66b733c6de899ade6ba47c71c0e710#diff-21b2361c0ea031da22f3680c5912c17ccc3d86c5bad901dd533f004cde3efa32 of https://github.com/matrix-org/matrix-react-sdk/pull/4159.

Reply:
I'm afraid I don't understand your answer. I think you are explaining why `.mx_TopUnreadMessagesBar_scrollUp` and `.mx_TopUnreadMessagesBar_markAsRead` are *combined*, but I didn't ask that. I asked why they are *nested* (inside `.mx_TopUnreadMessagesBar`).